selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubServiceClient.java

```diff
@@ -203,4 +204,24 @@ private String createBasicAuthHeader(String username, String password)
     String base64 = DatatypeConverter.printBase64Binary(nameAndPass);
     return "Basic " + base64;
   }
+
+  public String getUserPublicEmail(String username, String password) throws Exception {
+    String url = "https://api.github.com/user/public_emails";
+    HttpJsonResponse response =
+        requestFactory
+            .fromUrl(url)
+            .useGetMethod()
+            .setAuthorizationHeader(createBasicAuthHeader(username, password))
+            .request();
+
+    @SuppressWarnings("unchecked")
+    List<Map<String, String>> properties =
+        response.as(List.class, new TypeToken<List<Map<String, String>>>() {}.getType());
+
+    return properties
+        .stream()
+        .filter(map -> map.containsKey("email"))
+        .map(map -> map.get("email"))
+        .collect(Collectors.joining());
```

Comment:
What is the propose of joining() here? Did you try to test this method with github account with several public emails?

Reply:
I would recommend to get first email from the map instead of joining all of them as a less error-prone variant.